engine.py

```diff
@@ -23,200 +19,285 @@
 import nukescripts
 
 class NukeEngine(tank.platform.Engine):
+    """
+    An engine that supports Nuke 6.3v5+ and Hiero 9.0v*+.
+    """
+    # Define the different areas where menu events can occur in Hiero.
+    (HIERO_BIN_AREA, HIERO_SPREADSHEET_AREA, HIERO_TIMELINE_AREA) = range(3)
+
+    def __init__(self, *args, **kwargs):
+        self._hiero_enabled = nuke.env.get("hiero")
+        self._ui_enabled = nuke.env.get("gui")
+        super(NukeEngine, self).__init__(*args, **kwargs)
+
+    #####################################################################################
+    # Properties
 
-    ##########################################################################################
-    # init
+    @property
+    def ui_enabled(self):
```

Comment:
i replied to one of the earlier comments - but probably good to add it here just so it doesn't get "folded away" by github: The `has_ui` is part of the official engine interface so we cannot easily change that method! It's a shame beacuse `ui_enabled` works nicely with `hiero_enabled` (and `has_hiero` sounds really weird to me!). But we probably need to keep `has_ui` for consistency and backwards compatibility....